server/src/test/java/org/elasticsearch/index/codec/tsdb/ES87TSDBDocValuesEncoderTests.java

```diff
@@ -260,12 +260,59 @@ public void testEncodeOrdinalsNoRepetitions() throws IOException {
         doTestOrdinals(arr, 113);
     }
 
+    public void testEncodeOrdinalsBitPack3Bits() throws IOException {
+        long[] arr = new long[blockSize];
+        Arrays.fill(arr, 4);
+        for (int i = 0; i < 4; i++) {
+            arr[i] = i;
+        }
+        doTestOrdinals(arr, 49);
+    }
+
+    public void testEncodeOrdinalsCycle2() throws IOException {
+        long[] arr = new long[blockSize];
+        Arrays.setAll(arr, i -> i % 2);
+        doTestOrdinals(arr, 3);
```

Comment:
should we also assert that we actually detected the cycle, and that we did not use bit packing in that case?

One way to do that would be to make sure that we encoded the data on less than 1 bit per value, ie. the output is encoded on less than 128 bits.

Reply:
The second argument is the number of bytes. So this assets the whole array of 128 longs has been encoded with 3 bytes.

Reply:
Ohhh you're right, sorry for the noise!